src/main/java/com/jfrog/ide/idea/scan/SourceCodeScannerManager.java

```diff
@@ -96,14 +96,26 @@ private List<String> getSkippedFoldersPatterns() {
         return skippedFoldersPatterns;
     }
 
+    /**
+     * Convert {@link JFrogSecurityWarning}s to a list of {@link FileTreeNode}s with source code issues.
+     * Also, update {@link VulnerabilityNode}s with their matching source code issues.
```

Comment:
Since the name of the function, `groupResultsToFileTreeNodes` has nothing to do with updating, I recommend moving the update logic to another function.

In addition, the comment begins with the verb 'convert..', but the function name implies grouping. Anything that forces you to check the function body should be avoided.

Reply:
Moving the update logic to another method will make the code less efficient.
I made some changes to make this method more readable, changed its name, and fixed its javadoc.